File: web/src/types/templateTag.ts
/**
 * 方案模板标签。
 */
export interface TemplateTag {
  tagId: string
  tagName: string
  sortOrder: number
  enabled: boolean
  createdAt?: string
}

/**
 * 模板标签创建/更新请求。
 */
export interface TemplateTagPayload {
  tagName: string
  sortOrder?: number | null
  enabled?: boolean | null
}

File: server/src/main/java/com/rsdp/entity/TemplateTag.java
package com.rsdp.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 方案模板标签实体（受控字典；scheme.template_tags 存名称 JSON，以名称为业务键）。
 */
@Data
@TableName("template_tag")
public class TemplateTag {

    @TableId
    private String tagId;

    private String tagName;

    private Integer sortOrder;

    private Boolean enabled;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
